qemu/tests/pvpanic.py

```diff
@@ -0,0 +1,131 @@
+        status, output = session.cmd_status_output(cmd, timeout)
+        if status:
+            test.error("Command '%s' failed, status: %s, output: %s" %
+                       (cmd, status, output))
+
+
+def check_qmp_events(vm, event_name, timeout=600):
+    """
+    Check whether certain qmp event appeared in vm.
+
+    :param vm: target virtual machine
+    :param event_name: target event name, such as 'GUEST_PANICKED'
+    :param timeout: check time
+    """
+
+    end_time = time.time() + timeout
+    qmp_monitors = vm.get_monitors_by_type("qmp")
+    logging.info("Try to get qmp events %s in %s seconds!" %
+                 (event_name, timeout))
+    while time.time() < end_time:
+        for monitor in qmp_monitors:
+            event = monitor.get_event(event_name)
+            if event:
+                logging.info("Receive qmp %s event notification",
+                             event_name)
+                return True
+        time.sleep(5)
+    return False
+
+
+def trigger_crash(vm, params):
+    """
+    Trigger system crash with certain method
+
+    :param vm: target vm
+    :parma params: test params
+    """
+    # to do: will add other crash method
+    crash_method = params.get("crash_method")
+    if crash_method == "nmi":
+        vm.monitor.nmi()
+    else:
+        logging.error("Crash trigger mether %s not supported, "
+                      "please check cfg file for mistake.", crash_method)
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    pvpanic test
+
+    1) Log into the guest
+    2) Check if the driver is installed and verified (only for win)
+    3) Stop kdump service and modify unknown_nmi_panic(for linux)
+       or modify register value(for win)
+    4) Trigger a crash by nmi
+    5) Check the event in qmp
+    6) Check guest status
+
+    :param test: kvm test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    timeout = int(params.get("timeout", 360))
+    event_check = params.get("event_check", "GUEST_PANICKED")
+    guest_status = params.get("guest_status", "guest-panicked")
+    error_context.context("Boot guest with pvpanic device", logging.info)
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    session = vm.wait_for_login(timeout=timeout)
+    try:
+        if params.get("os_type") == "windows":
+            error_context.context("Check if the driver is installed and "
+                                  "verified", logging.info)
+            driver_name = params.get("driver_name", "pvpanic")
+            status, output = utils_test.qemu.check_win_driver(session,
+                                                              driver_name,
+                                                              timeout)
+            if status:
+                test.error("%s driver is not running, detailed info is %s" %
+                           (driver_name, output))
```

Comment:
when driver is not installed,the output is always empty.
I don't suggest you print empty output

Reply:
After discussed, xiagao will update #1201. 
Thanks